src/metering/metering.rs

```diff
@@ -76,37 +80,62 @@ pub(crate) struct MeteringRecorderSender {
     evaluation_event_sender: mpsc::Sender<EvaluationEvent>,
 }
 
-impl MeteringRecorderSender {
-    /// Record the evaluation of a property, for eventual transmission to the server.
-    pub fn record_property_evaluation(
+pub(crate) trait MeteringSubject {
+    fn record_evaluation(
```

Comment:
I wonder if we should take the `Entity` and `Segment` objects here and make sure that we are using `segment_id` and `entity_id` instead of relying on the caller. wdyt?